services/tests/test_test_results.py

```diff
@@ -239,3 +242,39 @@ def test_notify_fail_no_pull(
 
     notification_result = tn.notify()
     assert notification_result == NotifierResult.NO_PULL
+
+
+@pytest.mark.parametrize(
+    "config,feature_flag,private,plan,ex_result",
+    [
+        (False, True, False, "users-inappm", False),
+        (True, True, True, "users-basic", True),
+        (True, False, False, "users-basic", True),
+        (True, False, True, "users-basic", False),
+        (True, False, False, "users-inappm", True),
+        (True, False, True, "users-inappm", True),
+    ],
+)
+def test_should_do_flake_detection(
+    dbsession, mocker, config, feature_flag, private, plan, ex_result
+):
+    repo = RepositoryFactory()
+    dbsession.add(repo)
+    dbsession.flush()
+
+    if private:
+        repo.private = True
+    else:
+        repo.private = False
```

Comment:
```suggestion
    repo.private = private
```

or maybe even put that into the `RepositoryFactory` invocation?